### PR TITLE
Avoid ReplaceView in TextInputViewManager

### PR DESCRIPTION
**Pull request by rozele**
### Type of Change
_Erase all that don't apply._
- Bug fix (non-breaking change which fixes an issue)

### Why
Since we now have props in the CreateView method for core view managers, we can avoid the creation of a TextBox initially in TextInputViewManager when "secureTextEntry" is set to true.

Towards #7143

### What
Checks the props passed into CreateViewCore for `secureTextEntry`  == `true` and conditionally creates a `PasswordBox`, rather than a `TextBox`.

## Testing
Ran TextInput example in RNTester and everything works fine.

_Optional_: Describe the tests that you ran locally to verify your changes.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10356)

### CODE REVIEW

**Review by jonthysell — APPROVED** [review]

File: packages/e2e-test-app/test/__snapshots__/LegacyControlStyleTest.test.ts.snap [hunk 1]
```diff
@@ -1761,7 +1761,7 @@ Object {
               "Clip": null,
               "CornerRadius": "0,0,0,0",
               "FlowDirection": "LeftToRight",
-              "Foreground": "#FFFFFFFF",
+              "Foreground": "#FF000000",
```
Comment (jonthysell) [inline packages/e2e-test-app/test/__snapshots__/LegacyControlStyleTest.test.ts.snap:1764]: Is there a reason why this changes here?

File: packages/e2e-test-app/test/__snapshots__/LegacyControlStyleTest.test.ts.snap [hunk 2]
```diff
@@ -1812,7 +1812,7 @@ Object {
                           "Clip": "Windows.UI.Xaml.Media.RectangleGeometry",
                           "CornerRadius": "0,0,0,0",
                           "FlowDirection": "LeftToRight",
-                          "Foreground": "#FFFFFFFF",
+                          "Foreground": "#FF000000",
```
Comment (jonthysell) [inline packages/e2e-test-app/test/__snapshots__/LegacyControlStyleTest.test.ts.snap:1815]: Is there a reason why this changes here?

File: vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp [hunk 3]
```diff
@@ -477,6 +483,7 @@ void TextInputShadowNode::updateProperties(winrt::Microsoft::ReactNative::JSValu
   auto hasKeyDownEvents = false;
 
   auto markDirty = false;
+  auto isTextBox = IsTextBox();
   for (auto &pair : props) {
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp:487]: Should we be making sure that we process `secureTextEntry` first before any other changes?

Comment (rozele) [reply vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp:487]: We update the local value here: https://github.com/microsoft/react-native-windows/pull/10356/files#diff-d04c9c359dcc99f741f5d27e172658539f20871b36d8bdae9cae53aa028f0458R537

which is what we were doing before when updating the field on the shadow node, so no net new bugs should be introduced here.
